From 5e06792ded468da3ca54e97040548d64fb4a32d2 Mon Sep 17 00:00:00 2001
From: Andrei Gherzan <andrei@balena.io>
Date: Wed, 27 Oct 2021 17:18:57 +0200
Subject: [PATCH] core: Don't redirect stdio to null when running in container

Upstream-status: Rejected [https://github.com/systemd/systemd/pull/8220]
Signed-off-by: Andrei Gherzan <andrei@balena.io>
---
 src/core/main.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/core/main.c b/src/core/main.c
index b32a19a1d8..072f803e03 100644
--- a/src/core/main.c
+++ b/src/core/main.c
@@ -2745,7 +2745,8 @@ int main(int argc, char *argv[]) {
                  * they never log into the console. */
                 log_show_color(colors_enabled());
 
-                r = make_null_stdio();
+                if (detect_container() <= 0)
+                        r = make_null_stdio();
                 if (r < 0)
                         log_warning_errno(r, "Failed to redirect standard streams to /dev/null, ignoring: %m");
 
-- 
2.17.1

