From 4b4337cc9d7c13be578ff54457fb0ca1ac4273df Mon Sep 17 00:00:00 2001
From: Brennen Bearnes <bbearnes@gmail.com>
Date: Thu, 3 Sep 2015 19:45:20 -0600
Subject: [PATCH] ft6x06_ts:  move suspend/resume into driver portion of struct

---
 drivers/input/touchscreen/ft6x06_ts.c | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/drivers/input/touchscreen/ft6x06_ts.c b/drivers/input/touchscreen/ft6x06_ts.c
index 3aed5a7..ba65b48 100644
--- a/drivers/input/touchscreen/ft6x06_ts.c
+++ b/drivers/input/touchscreen/ft6x06_ts.c
@@ -464,17 +464,16 @@ static struct of_device_id ft6x06_match_table[] = {
 #define ft6x06_match_table NULL
 #endif
 
-
 static struct i2c_driver ft6x06_ts_driver = {
 	.probe = ft6x06_ts_probe,
 	.remove = ft6x06_ts_remove,
 	.id_table = ft6x06_ts_id,
-	.suspend = ft6x06_ts_suspend,
-	.resume = ft6x06_ts_resume,
 	.driver = {
 		.name = FT6X06_NAME,
 		.owner = THIS_MODULE,
 		.of_match_table=ft6x06_match_table,
+		.suspend = ft6x06_ts_suspend,
+		.resume = ft6x06_ts_resume,
 	},
 };
 
-- 
2.6.2

