From c736b5d679c7a1c6635ca908b27f962b107719a5 Mon Sep 17 00:00:00 2001
From: Brennen Bearnes <bbearnes@gmail.com>
Date: Thu, 3 Sep 2015 19:50:40 -0600
Subject: [PATCH] ft6x06: s/drv/dev

---
 drivers/input/touchscreen/ft6x06_ts.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/input/touchscreen/ft6x06_ts.c b/drivers/input/touchscreen/ft6x06_ts.c
index ba65b48..f4eba6b 100644
--- a/drivers/input/touchscreen/ft6x06_ts.c
+++ b/drivers/input/touchscreen/ft6x06_ts.c
@@ -423,7 +423,7 @@ static int ft6x06_ts_suspend(struct device *dev, pm_message_t mesg)
 
 static int ft6x06_ts_resume(struct device *dev)
 {
-	struct ft6x06_ts_data *ts = dev_get_drvdata(drv);
+	struct ft6x06_ts_data *ts = dev_get_drvdata(dev);
 	dev_dbg(&ts->client->dev, "[FTS]ft6x06 resume.\n");
 	if (gpio_is_valid(ts->pdata->reset_gpio)) {
 		gpio_set_value(ts->pdata->reset_gpio, 0);
-- 
2.6.2

