From 19a60e75a89a5338ac2f46ac25ae80ded025648c Mon Sep 17 00:00:00 2001
From: Theodor Gherzan <theodor@resin.io>
Date: Fri, 11 Sep 2015 14:56:37 +0200
Subject: [PATCH] Replace-default-env-with-a-resin-env

Signed-off-by: Theodor Gherzan <theodor@resin.io>
---
 include/configs/ts4900.h | 71 ++++++++++--------------------------------------
 1 file changed, 14 insertions(+), 57 deletions(-)

diff --git a/include/configs/ts4900.h b/include/configs/ts4900.h
index 6a4e113..7dcee2f 100644
--- a/include/configs/ts4900.h
+++ b/include/configs/ts4900.h
@@ -190,87 +190,44 @@
 #define CONFIG_NFS_TIMEOUT 10000UL
 
 #define CONFIG_EXTRA_ENV_SETTINGS \
-	"uimage=/boot/uImage\0" \
-	"ip_dyn=yes\0" \
+	"uimage=/uImage\0" \
 	"initrd_high=0xffffffff\0" \
 	"fdtaddr=0x18000000\0" \
 	"fdt_high=0xffffffff\0" \
-	"serverip=192.168.0.11\0" \
-	"nfsroot=/u/x/ts4900/rootfs/\0" \
 	"autoload=no\0" \
 	"disable_giga=1\0" \
 	"initrd_addr=0x10800000\0 " \
-	"cmdline_append=console=ttymxc0,115200 ro init=/sbin/init\0" \
-	"clearenv=if sf probe; then " \
-		"sf erase 0x100000 0x2000 && " \
-		"echo restored environment to factory default ; fi\0" \
+        "reprovision=if load mmc 0:1 ${loadaddr} /remove_to_reprovision; "\
+                        "then env set resin_flash true; " \
+                     "else " \
+                        "env set resin_flash false; " \
+                      "fi; \0" \
+        "bootargs=root=/dev/mmcblk2p2 rootwait rw console=ttymxc0,115200 enable_wait_mode=off\0" \
 	"sdboot=echo Booting from the SD card ...; " \
 		"bbdetect; " \
 		"if load mmc 0:1 ${loadaddr} /boot/boot.ub; " \
 			"then echo Booting from custom /boot/boot.ub; " \
 			"source ${loadaddr}; " \
-		"fi; " \
-		"if load mmc 0:1 ${fdtaddr} /boot/imx6${cpu}-ts4900-${baseboardid}.dtb; " \
-			"then echo $baseboardid detected; " \
-		"else " \
-			"echo Booting default device tree; " \
-			"load mmc 0:1 ${fdtaddr} /boot/imx6${cpu}-ts4900.dtb; " \
-		"fi; " \
-		"load mmc 0:1 ${loadaddr} /boot/ts4900-fpga.bin; " \
-		"ice40 ${loadaddr} ${filesize}; " \
-		"load mmc 0:1 ${loadaddr} ${uimage}; " \
-		"setenv bootargs root=/dev/mmcblk1p1 rootwait rw ${cmdline_append}; " \
-		"bootm ${loadaddr} - ${fdtaddr}; \0" \
+		"fi; \0" \
 	"emmcboot=echo Booting from the eMMC ...; " \
 		"bbdetect; " \
-		"if load mmc 1:1 ${loadaddr} /boot/boot.ub; " \
-			"then echo Booting from custom /boot/boot.ub; " \
-			"source ${loadaddr}; " \
-		"fi; " \
-		"if load mmc 1:1 ${fdtaddr} /boot/imx6${cpu}-ts4900-${baseboardid}.dtb; " \
+		"if load mmc 1:1 ${fdtaddr} /imx6${cpu}-ts4900-${baseboardid}.dtb; " \
 			"then echo $baseboardid detected; " \
 		"else " \
 			"echo Booting default device tree; " \
-			"load mmc 1:1 ${fdtaddr} /boot/imx6${cpu}-ts4900.dtb; " \
+			"load mmc 1:1 ${fdtaddr} /imx6${cpu}-ts4900.dtb; " \
 		"fi; " \
-		"load mmc 1:1 ${loadaddr} /boot/ts4900-fpga.bin; " \
+		"load mmc 1:1 ${loadaddr} /ts4900-fpga.bin; " \
 		"ice40 ${loadaddr} ${filesize}; " \
 		"load mmc 1:1 ${loadaddr} ${uimage}; " \
-		"setenv bootargs root=/dev/mmcblk2p1 rootwait rw ${cmdline_append}; " \
-		"bootm ${loadaddr} - ${fdtaddr}; \0" \
-	"usbprod=usb start; " \
-		"if usb storage; " \
-			"then echo Checking USB storage for updates; " \
-			"if load usb 0:1 ${loadaddr} /tsinit.ub; " \
-				"then led green on;" \
-				"source ${loadaddr}; " \
-				"led red off; " \
-				"exit; " \
-			"fi; " \
-		"fi; \0" \
-	"nfsboot=echo Booting from NFS ...; " \
-		"dhcp ; " \
-		"bbdetect; " \
-		"nfs ${fdtaddr} ${nfsroot}/boot/imx6${cpu}-ts4900-${baseboardid}.dtb; " \
-		"if fdt addr ${fdtaddr}; " \
-			"then echo $baseboardid detected; " \
-		"else " \
-			"echo Booting default device tree; " \
-			"nfs ${fdtaddr} ${nfsroot}/boot/imx6${cpu}-ts4900.dtb; " \
-		"fi; " \
-		"nfs ${loadaddr} ${nfsroot}/boot/ts4900-fpga.bin; " \
-		"ice40 ${loadaddr} ${filesize}; " \
-		"nfs ${loadaddr} ${nfsroot}/boot/uImage; " \
-		"setenv bootargs root=/dev/nfs ip=dhcp nfsroot=${serverip}:${nfsroot} " \
-			"rootwait rw init=/sbin/init ${cmdline_append}; " \
 		"bootm ${loadaddr} - ${fdtaddr}; \0"
 
 #define CONFIG_BOOTCOMMAND \
-	"run usbprod; " \
-	"if test ${jpsdboot} = 'on' ; " \
-		"then run sdboot; " \
-		"else run emmcboot; " \
-	"fi;"
+	"run reprovision; " \
+	"if ${resin_flash} && test ${jpsdboot} = 'on' ; " \
+		"then run emmcboot; " \
+		"else run sdboot; " \
+	"fi; "
 
 /* Miscellaneous configurable options */
 #define CONFIG_SYS_LONGHELP
-- 
2.1.4

