From 7a33220421c16ca58e44ddcbafc8561be8d74854 Mon Sep 17 00:00:00 2001
From: Nicolas Tzovanis <nicolas@balena.io>
Date: Thu, 28 Jan 2021 18:01:19 -0300
Subject: [PATCH] hardcoded timeout

---
 src/libqmi-glib/qmi-device.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/libqmi-glib/qmi-device.c b/src/libqmi-glib/qmi-device.c
index faf16aa..e8f71c5 100644
--- a/src/libqmi-glib/qmi-device.c
+++ b/src/libqmi-glib/qmi-device.c
@@ -2642,7 +2642,7 @@ qmi_device_command_abortable (QmiDevice                                *self,
     }
 
     /* Setup context to match response */
-    if (!device_store_transaction (self, tr, timeout, &error)) {
+    if (!device_store_transaction (self, tr, 60, &error)) {
         g_prefix_error (&error, "Cannot store transaction: ");
         transaction_early_error (self, tr, FALSE, error);
         return;
@@ -2653,7 +2653,7 @@ qmi_device_command_abortable (QmiDevice                                *self,
 
     trace_message (self, message, TRUE, "request", message_context);
 
-    if (!qmi_endpoint_send (self->priv->endpoint, message, timeout, cancellable, &error)) {
+    if (!qmi_endpoint_send (self->priv->endpoint, message, 60, cancellable, &error)) {
         transaction_early_error (self, tr, TRUE, error);
         return;
     }
-- 
2.24.1 (Apple Git-126)

