From 659b15313b9ca8c7f3f6e523e0d3fd696243ff8b Mon Sep 17 00:00:00 2001
From: Jan Janssen <medhefgo@web.de>
Date: Thu, 8 Mar 2018 16:44:17 +0100
Subject: [PATCH] umount: Fix memory leak

Upstream-Status: Backport
Signed-off-by: Florin Sarbu <florin@balena.io>
---
 src/core/umount.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/core/umount.c b/src/core/umount.c
index ff3e637..8ba4179 100644
--- a/src/core/umount.c
+++ b/src/core/umount.c
@@ -61,6 +61,8 @@ static void mount_point_free(MountPoint **head, MountPoint *m) {
         LIST_REMOVE(mount_point, *head, m);
 
         free(m->path);
+        free(m->options);
+        free(m->type);
         free(m);
 }
 
-- 
2.7.4

