From e783b4902f387640bba12496936d01e967545c3c Mon Sep 17 00:00:00 2001
From: Jan Janssen <medhefgo@web.de>
Date: Thu, 8 Mar 2018 18:51:13 +0100
Subject: [PATCH] umount: Don't bother remounting api and ro filesystems
 read-only

Upstream-Status: Backport
Signed-off-by: Florin Sarbu <florin@balena.io>
---
 src/core/umount.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/core/umount.c b/src/core/umount.c
index 40153de..f9b4a21 100644
--- a/src/core/umount.c
+++ b/src/core/umount.c
@@ -147,6 +147,8 @@ static int mount_points_list_get(MountPoint **head) {
                  * hang because of the network being down. */
                 m->try_remount_ro = detect_container() <= 0 &&
                                     !fstype_is_network(type) &&
+                                    !fstype_is_api_vfs(type) &&
+                                    !fstype_is_ro(type) &&
                                     !fstab_test_yes_no_option(options, "ro\0rw\0");
 
                 if (m->try_remount_ro) {
-- 
2.7.4

