Configure DS23 as LED driven by gpio-leds

Upstream-Status: Inappropriate [configuration]

Signed-off-by: Andrei Gherzan <andrei@resin.io>

Index: 1.0-r0/zc702/zc702-zynq7.dts
===================================================================
--- 1.0-r0.orig/zc702/zc702-zynq7.dts
+++ 1.0-r0/zc702/zc702-zynq7.dts
@@ -2,4 +2,12 @@
 /include/ "zynq7-base.dtsi"
 /include/ "zc702-zynq7-board.dtsi"
 / {
+    leds {
+	compatible = "gpio-leds";
+
+	ds23 {
+		gpios = <&ps7_gpio_0 10 0>;
+		default-state = "off";
+	};
+    };
 } ;
\ No newline at end of file
