From 08556ebaead2aee9557e00da205f7525d2295404 Mon Sep 17 00:00:00 2001
From: Andrei Gherzan <andrei@balena.io>
Date: Fri, 1 Jun 2018 15:09:01 +0100
Subject: [PATCH] Install link in /usr/bin to match other providers

Signed-off-by: Andrei Gherzan <andrei@balena.io>
Upstream-status: Inappropiate [Configuration]

---
 coreutils/link.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/coreutils/link.c b/coreutils/link.c
index d8d583b..664c7d1 100644
--- a/coreutils/link.c
+++ b/coreutils/link.c
@@ -11,7 +11,7 @@
 //config:	help
 //config:	link creates hard links between files.
 
-//applet:IF_LINK(APPLET_NOFORK(link, link, BB_DIR_BIN, BB_SUID_DROP, link))
+//applet:IF_LINK(APPLET_NOFORK(link, link, BB_DIR_USR_BIN, BB_SUID_DROP, link))
 
 //kbuild:lib-$(CONFIG_LINK) += link.o
 
