From 5ed5d5ed2f37e3d0c213fbbbfa7b24342a2ae872 Mon Sep 17 00:00:00 2001
From: Eric Bischoff <ebischoff@suse.com>
Date: Wed, 12 Feb 2020 15:34:41 +0100
Subject: [PATCH] Do not cache responses generated locally

---
 avahi-core/server.c | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/avahi-core/server.c b/avahi-core/server.c
index a2580e38a349..a44fb5754d3e 100644
--- a/avahi-core/server.c
+++ b/avahi-core/server.c
@@ -676,9 +676,11 @@ static void handle_response_packet(AvahiServer *s, AvahiDnsPacket *p, AvahiInter
         if (!avahi_key_is_pattern(record->key)) {
 
             if (handle_conflict(s, i, record, cache_flush)) {
-                if (!from_local_iface && !avahi_record_is_link_local_address(record))
-                    reflect_response(s, i, record, cache_flush);
-                avahi_cache_update(i->cache, record, cache_flush, a);
+                if (!from_local_iface) {
+                    if (!avahi_record_is_link_local_address(record))
+                        reflect_response(s, i, record, cache_flush);
+                    avahi_cache_update(i->cache, record, cache_flush, a);
+                }
                 avahi_response_scheduler_incoming(i->response_scheduler, record, cache_flush);
             }
         }
